localstack/services/sns/publisher.py

```diff
@@ -0,0 +1,946 @@
+            values = ast.literal_eval(val)
+            for value in values:
+                for condition in conditions:
+                    if self._evaluate_condition(value, condition, message_attributes, criteria):
+                        return True
+        else:
+            for condition in conditions:
+                value = val or None
+                if self._evaluate_condition(value, condition, message_attributes, criteria):
+                    return True
+
+        return False
+
+    def _evaluate_condition(self, value, condition, message_attributes, criteria):
+        if type(condition) is not dict:
+            return value == condition
+        elif condition.get("exists") is not None:
+            return self._evaluate_exists_condition(
+                condition.get("exists"), message_attributes, criteria
+            )
+        elif value is None:
+            # the remaining conditions require the value to not be None
+            return False
+        elif condition.get("anything-but"):
+            return value not in condition.get("anything-but")
+        elif condition.get("prefix"):
+            prefix = condition.get("prefix")
+            return value.startswith(prefix)
+        elif condition.get("numeric"):
+            return self._evaluate_numeric_condition(condition.get("numeric"), value)
+        return False
+
+    @staticmethod
+    def _evaluate_numeric_condition(conditions, value):
+        try:
+            # try if the value is numeric
+            value = float(value)
+        except ValueError:
+            # the value is not numeric, the condition is False
+            return False
+
+        for i in range(0, len(conditions), 2):
+            operator = conditions[i]
+            operand = float(conditions[i + 1])
+
+            if operator == "=":
+                if value != operand:
+                    return False
+            elif operator == ">":
+                if value <= operand:
+                    return False
+            elif operator == "<":
+                if value >= operand:
+                    return False
+            elif operator == ">=":
+                if value < operand:
+                    return False
+            elif operator == "<=":
+                if value > operand:
+                    return False
+
+        return True
+
+    @staticmethod
+    def _evaluate_exists_condition(conditions, message_attributes, criteria):
+        # support for exists: false was added in april 2021
+        # https://aws.amazon.com/about-aws/whats-new/2021/04/amazon-sns-grows-the-set-of-message-filtering-operators/
+        if conditions:
+            return message_attributes.get(criteria) is not None
+        else:
+            return message_attributes.get(criteria) is None
+
+
+class PublishDispatcher:
+    _http_publisher = HttpTopicPublisher()
+    topic_notifiers = {
+        "http": _http_publisher,
+        "https": _http_publisher,
```

Comment:
nit: i think consistency is better here than saving the one extra call:

```suggestion
    topic_notifiers = {
        "http": HttpTopicPublisher(),
        "https": HttpTopicPublisher(),
```